utils/inpsSoapClient.ts

```diff
@@ -0,0 +1,110 @@
+  </con:ConsultazioneSogliaIndicatoreRequest>
+</soapenv:Body>
+</soapenv:Envelope>`;
+
+const INPS_NAMESPACE = "http://inps.it/ConsultazioneISEE";
+
+export interface ISoapClientAsync {
+  ConsultazioneSogliaIndicatore: (
+    params: ConsultazioneSogliaIndicatoreInput
+  ) => TaskEither<Error, ConsultazioneSogliaIndicatoreResponse>;
+}
+
+export function createClient(
+  INPS_SERVICE_HOST: NonEmptyString,
+  endpoint: NonEmptyString
+): ISoapClientAsync {
+  return {
+    ConsultazioneSogliaIndicatore: (
+      params: ConsultazioneSogliaIndicatoreInput
+    ) => {
+      return tryCatch(async () => {
+        const requestPayload = getSOAPRequest(
+          params.DataValidita || format(new Date(), "yyyy-MM-dd"),
+          params.CodiceFiscale,
+          params.FornituraNucleo,
+          params.CodiceSoglia
+        );
+        const response = await fetch(`${INPS_SERVICE_HOST}${endpoint}`, {
+          body: requestPayload,
+          method: "POST"
+        });
+        if (response.status === 200) {
+          const responseBody = await response.text();
+          const xmlDocument = new DOMParser().parseFromString(
+            responseBody,
+            "text/xml"
+          );
+          return fromNullable(
+            new Error("Missing ConsultazioneSogliaIndicatoreResult")
+          )(
+            xmlDocument
+              .getElementsByTagNameNS(
+                INPS_NAMESPACE,
+                "ConsultazioneSogliaIndicatoreResult"
+              )
+              .item(0)
+          )
+            .map(_ => ({
+              DataPresentazioneDSU: _.getAttribute("DataPresentazioneDSU"),
+
+              ProtocolloDSU: _.getAttribute("ProtocolloDSU"),
+
+              SottoSoglia: _.getAttribute("SottoSoglia"),
+
+              TipoIndicatore: _.getAttribute("TipoIndicatore")
+            }))
+            .chain(_ => {
+              return ConsultazioneSogliaIndicatoreResponse.decode({
+                ..._,
+                Componente: Array.from(
+                  xmlDocument.getElementsByTagNameNS(
+                    INPS_NAMESPACE,
+                    "Componente"
+                  )
+                ).map(familyMemberElement => ({
+                  CodiceFiscale: familyMemberElement.getAttribute(
+                    "CodiceFiscale"
+                  ),
+                  Cognome: familyMemberElement.getAttribute("Cognome"),
+                  Nome: familyMemberElement.getAttribute("Nome")
+                }))
+              }).mapLeft(error => {
+                return new Error(
+                  `Unexpected response: [Error: ${readableReport(error)}]`
+                );
+              });
+            })
+            .fold(
+              _ => Promise.reject(_),
+              _ => Promise.resolve(_)
+            );
+        }
+        return Promise.reject(new Error("Unexpected statusCode response"));
```

Comment:
maybe I'm wrong but it looks like this reject here is called unconditionally

Reply:
is called if [this](https://github.com/pagopa/io-functions-bonus/pull/7/files#diff-d664327273703ce97d04e0c72db5237bR55) if is false